File: Data Structures & Algorithms/depth-of-binary-tree/submission-0.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

public class Solution {
    public int MaxDepth(TreeNode root) {
                if (root is null)
        {
            return 0;
        }
        var queue = new Queue<TreeNode>();
        queue.Enqueue(root);
        var levels = 0;
        while (queue.Count > 0)
        {
            var currentSize = queue.Count;
            for (var i = 0; i < currentSize; i++)
            {
                var node = queue.Dequeue();
                if (node.left is not null)
                    queue.Enqueue(node.left);
                if (node.right is not null)
                    queue.Enqueue(node.right);
            }
            levels++;
        }
        return levels;
    }
}

File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
        if not root:
            return 0

        q = deque([root])
        pq = []

        while q:
            for i in range(len(q)):
                node = q.popleft()
                pq.append(node.val)
                if node.left:
                    q.append(node.left)
                if node.right:
                    q.append(node.right)

        heapq.heapify(pq)
        for _ in range(k - 1):
            print(heapq.heappop(pq))
        item = heapq.heappop(pq)
        return item